crate/cli/src/actions/findex/insert_or_delete.rs

```diff
@@ -58,48 +60,67 @@ impl InsertOrDeleteAction {
         Ok(KeywordToDataSetsMap(csv_in_memory))
     }
 
-    /// Insert or delete indexes
     async fn insert_or_delete(
         &self,
         rest_client: &FindexRestClient,
         is_insert: bool,
-    ) -> CliResult<Keywords> {
-        let keywords_indexed_value = self.to_keywords_indexed_value_map()?;
-        let findex: Findex<WORD_LENGTH, Value, String, FindexRestClient> =
+    ) -> CliResult<String> {
+        let bindings = self.to_indexed_value_keywords_map()?;
+
         // cloning will be eliminated in the future, cf https://github.com/Cosmian/findex-server/issues/28
+        let findex = Arc::<Findex<WORD_LENGTH, Value, String, FindexRestClient>>::new(
             rest_client.clone().instantiate_findex(
                 self.findex_parameters.index_id,
                 &self.findex_parameters.seed()?,
-            )?;
-        for (key, value) in keywords_indexed_value.iter() {
-            if is_insert {
-                trace!("Attempt to insert ...");
-                findex.insert(key, value.clone()).await
-            } else {
-                findex.delete(key, value.clone()).await
-            }?;
+            )?,
+        );
+        let semaphores = Arc::new(Semaphore::new(MAX_SEMAPHORES));
```

Comment:
A semaphore is a singular entity : no `s`.

There may be an issue here if someone inserts or deletes in parallel several big documents. We may decide not to attempt capturing all user behaviors no to add complexity. But this corner case should be documented somewhere.

Reply:
Also, I don't like the naming `MAX_SEMAPHORES` since as I said, a semaphore is a singular entity. `MAX_PERMITS` would be a better choice since it would echo the semaphore documentation:

```
    /// Creates a new semaphore with the initial number of permits.
```